app/javascript/mastodon/features/status/components/card.js

```diff
@@ -75,14 +79,25 @@ export default class Card extends React.PureComponent {
     );
   }
 
+  setRef = c => {
+    if (c) {
+      this.setState({ width: c.offsetWidth });
```

Comment:
So unfortunately `offsetWidth` triggers layout… but this is only for video embeds right? Basically YouTube and whatnot?

Reply:
Yes. YouTube, Vimeo, Soundcloud, Imgur, Twitch Clip, etc

Reply:
Mind you, StatusCard is only ever mounted from DetailedStatus!